packages/accounts/src/account-header.ts

```diff
@@ -0,0 +1,13 @@
+import { Address } from '@solana/addresses';
+import { LamportsUnsafeBeyond2Pow53Minus1 } from '@solana/rpc-types';
+
+/** The amount of bytes required to store the account header. */
+export const ACCOUNT_HEADER_SIZE = 128;
+
+/** Describe the generic account details applicable to every account. */
+export type AccountHeader = {
```

Comment:
Happy to rename that `BaseAccount`. 👌 

Are you happy with the constant name `ACCOUNT_HEADER_SIZE` or should we also rename that to `BASE_ACCOUNT_SIZE`? It basically refers to an account size minus its data size.

Reply:
I don't know. What's `ACCOUNT_HEADER_SIZE` for again? To know the offset if you want to skip to the good stuff and parse it yourself?

I feel like hard coding these values into the client and then letting them out into the world is dangerous. We're [struggling with this right now ](https://discord.com/channels/428295358100013066/1171197072045768946/1171256268384641054), because the size of the vote account is changing.

Reply:
Fair point! Let me try and add some context around this constant.

`ACCOUNT_HEADER_SIZE` is useful when trying to get the **total size or the total rent of more than one account**.

Say you know that a transaction will create 3 accounts on-chain: a `Mint` account, a `Token` account and a `Metadata` account. Assuming you know the size of each account, how do you figure out how much SOL your wallet should have to pay for that rent?

You'd think one could do:

```ts
const totalRent = await rpc.getRent(getMintSize() + getTokenSize() + getMetadataSize());
```

But **that will give you the wrong result** because `getRent` automatically add the `ACCOUNT_HEADER_SIZE` to the provided size. So the input of `getRent` is expected to be the account data of **a single account**.

To go around this, you could do three RPC calls:

```ts
const allRents = await Promise.all([
  rpc.getRent(getMintSize()),
  rpc.getRent(getTokenSize()),
  rpc.getRent(getMetadataSize()),
]);
const totalRent = allRents.reduce((all, one) => all + one, 0);
```

However, we can easily go from N to 1 RPC calls by doing some simple math that includes the size of the account header.

```ts
const accountHeaderRent = await rpc.getRent(0); // The dApp can even cache this.
const rentPerByte = accountHeaderRent / ACCOUNT_HEADER_SIZE;
const totalBytes = 3 * ACCOUNT_HEADER_SIZE + getMintSize() + getTokenSize() + getMetadataSize();
const totalRent = totalBytes * rentPerByte;
```

Additionally, `ACCOUNT_HEADER_SIZE` is used by Kinobi to compute the total amount of bytes created on-chain for a given instruction.

I agree that maintaining constant is a pain when they have the potential to evolve (especially at the different times on different clusters) but that's one of these constant that's so core that changing it would break a lot of things. For instance the implementation of the `getRent` RPC method would also need updating.

Reply:
inb4 someone suggests non-linear storage pricing and that breaks.

Reply:
Doesn't seem like we've landed on the header naming convention? Technically, the layout doesn't exactly represent header-body structure. I can imagine the "wtf is an account header" confusion.

I would vote for either `AccountBase` or `BaseAccount`.

Constant makes sense to me, though! Nifty for calculating account size. If it eventually breaks we can deprecate it I guess 😅

Reply:
I renamed `AccountHeader` and `ACCOUNT_HEADER_SIZE` into `BaseAccount` and `BASE_ACCOUNT_SIZE` respectively.

If we need to worry about non-linear storage pricing now, we have a lot more to worry than this simple constant.